### PR TITLE
adding create admin endpoint, protecting endpoint with jwt, validatin…

### PR DESCRIPTION
**Pull request by luis8alopez**
## Summary

- Adding create admin endpoint
- Using same schema validations that regular user
- Protecting endpoint with JWT 
- Validating that only admin users can consume endpoint
- Adding test

## Trello Card

https://trello.com/c/LLjPqamq/16-usuario-administrador

### CODE REVIEW

File: app/services/users.js [hunk 1]
```diff
@@ -27,3 +28,15 @@ exports.getUsers = async (page, limit) => {
     throw errors.databaseError('Error trying to fetch data from the DB');
   }
 };
+
+exports.createAdmin = async payload => {
+  try {
+    payload.role = admin_role;
```
Comment (dbenavidesv) [inline app/services/users.js]: `camelCase` here

File: config/development.js [hunk 2]
```diff
@@ -2,7 +2,8 @@ exports.config = {
   environment: 'development',
   common: {
     database: {
-      name: process.env.DB_NAME_DEV
+      name: process.env.DB_NAME_DEV,
+      admin_role: process.env.ADMIN_ROLE
```
Comment (dbenavidesv) [inline config/development.js]: I've noticed that you add the env variables in all environment files, this is not necessary, just add them in the index  of this folder. You should add them to a specific env file only when is needed, but generally adding them to the index is enough

File: migrations/migrations/20210331154903-users.js [hunk 3]
```diff
@@ -32,6 +32,9 @@ module.exports = {
       },
       updated_at: {
         type: Sequelize.DATE
+      },
+      role: {
+        type: Sequelize.STRING
```
Comment (dbenavidesv) [inline migrations/migrations/20210331154903-users.js]: You shouldn't modify existing migrations, if you need yo modify something in the db, you must create a new migration. For example, in this case you need to create a new migration that adds the new column `role`

File: app/services/users.js (hunk 1 above)
Comment (luis8alopez) [reply app/services/users.js]: But admin_role in this case is an env variable, does env variables have to be cameCase too?

File: app/services/users.js [hunk 4]
```diff
@@ -27,3 +28,15 @@ exports.getUsers = async (page, limit) => {
...
+    const userCreated = await User.upsert(payload, { returning: true });
+
+    return formatUserOutput(userCreated[0]);
```
Comment (dbenavidesv) [inline app/services/users.js]: same as in the previous pr, put the formatter in the controller

File: migrations/migrations/20210331154903-users.js (hunk 3 above)
Comment (dbenavidesv) [reply migrations/migrations/20210331154903-users.js:37]: Remember this part

**Review by dbenavidesv — APPROVED** [review]

**Pull request merged**